.github/workflows/build.yml

```diff
@@ -244,33 +244,57 @@ jobs:
   # Run Rust lints and tests
   rust-lint-and-tests:
     runs-on: ubuntu-20.04
+    env:
+      RUSTFLAGS: -Dwarnings
     strategy:
       matrix:
         manifest:
+          - cosmwasm/Cargo.toml
           - terra/Cargo.toml
           - sdk/rust/Cargo.toml
     steps:
       - name: Check out source
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-      - name: Install stable rust toolchain
-        uses: actions-rs/toolchain@v1
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
```

Comment:
I'm curious, does the upstream action now support the stable toolchain?
https://github.com/dtolnay/rust-toolchain/commit/84be2718c89917863ad106851300e80ecf0800f7 seems to only add stable as the default, but `actions-rs/toolchain` seems to support it as an additional flag: https://github.com/marketplace/actions/rust-toolchain#components

Reply:
I switched because actions-rs is [unmaintained](https://github.com/actions-rs/toolchain/issues/216).

In classic dtolnay fashion, there is a clever solution for non-stable toolchains.  If you want a specific toolchain, you simply check out the rust-toolchain action at the same branch.  For example, for the 1.64.0 release I would use `rust-toolchain@1.64.0`, which has [this code](https://github.com/dtolnay/rust-toolchain/blob/1.64.0/action.yml#L8-L12) instead:

```yaml
inputs:
  toolchain:
    description: Rust toolchain specification -- see https://rust-lang.github.io/rustup/concepts/toolchains.html#toolchain-specification
    required: false
    default: 1.64.0
```

For nightly, we have to use `rust-toolchain@master` and then explicitly list the toolchain version as one of the inputs.  This is what we're currently doing for the solana CI action.

Reply:
We might want to pin this to a hash for security purposes given this is an external dependency

Reply:
Done